File: backend/IDE.Common/ModelsDTO/DTO/Workspace/FileStructureDTO.cs
using IDE.Common.ModelsDTO.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace IDE.Common.ModelsDTO.DTO.Workspace
{
    public class FileStructureDTO
    {
        public FileStructureDTO()
        {
            NestedFiles = new List<FileStructureDTO> ();
        }
        public string Id { get; set; }
        public TreeNodeType Type { get; set; }
        public string Name { get; set; }
        public string Details { get; set; }
        public ICollection<FileStructureDTO> NestedFiles { get; set; }
    }
}

File: frontend/src/app/models/DTO/Workspace/fileStructureDTO.ts
/* Auto Generated */

import { TreeNodeType } from "./../../Enums/treeNodeType"
import { FileStructureDTO } from "./fileStructureDTO"

export interface FileStructureDTO {
    id: string;
    type: TreeNodeType;
    isOpened: boolean;
    name: string;
    size: number;
    details: string;
    nestedFiles: FileStructureDTO[];
}
